internal/pkg/runtime/engines/singularity/container.go

```diff
@@ -322,27 +281,32 @@ func (c *container) setupSessionLayout(system *mount.System) error {
 		}
 	}
 
-	if c.engine.EngineConfig.GetWritableImage() && !writableTmpfs {
-		imgObject, err := c.loadImage(c.engine.EngineConfig.GetImage(), true)
-		if err != nil {
-			return err
-		}
+	imgObject, err := c.loadImage(c.engine.EngineConfig.GetImage(), true)
+	if err != nil {
+		return err
+	}
 
+	if c.engine.EngineConfig.GetWritableImage() && !writableTmpfs {
+		sylog.Debugf("Image is writable, not attempting to use overlay or underlay\n")
 		if imgObject.Type == image.SIF {
-			err = c.setupWritableSIFImage(imgObject, overlayEnabled)
+			err = c.setupSIFOverlay(imgObject, c.engine.EngineConfig.GetWritableImage())
 			if err == nil {
 				return c.setupOverlayLayout(system, sessionPath)
 			}
 			sylog.Warningf("%s", err)
```

Comment:
I think we should make this more descriptive, in the same way as the other comment